### PR TITLE
Problem: v1.0.8 hotfix version is not released

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- cherry-pick ibc-go update and cosmos-sdk revert

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+*May 30, 2023*
+
+## v1.0.8
+
+- [#1038](https://github.com/crypto-org-chain/cronos/pull/1038) Update ibc-go to `v5.2.1`.
+- [#]() Revert accidental change in `v1.0.7` which causes `v1.0.7` to be non-deterministic.
```
Comment (yihuang) [inline CHANGELOG.md:8]:
```suggestion
- [#1052](https://github.com/crypto-org-chain/cronos/pull/1052) Revert accidental change in `v1.0.7` which causes `v1.0.7` to be non-deterministic.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+*May 30, 2023*
+
+## v1.0.8
+
+- [#1038](https://github.com/crypto-org-chain/cronos/pull/1038) Update ibc-go to `v5.2.1`.
+- [#1052](https://github.com/crypto-org-chain/cronos/pull/1052) Revert accidental change in `v1.0.7` which causes `v1.0.7` to be non-deterministic.
```
Comment (yihuang) [inline CHANGELOG.md:8]:
```suggestion
- [#1052](https://github.com/crypto-org-chain/cronos/pull/1052) Revert accidental breaking change in `v1.0.7`.
```

File: go.mod [hunk 3]
```diff
@@ -153,29 +154,29 @@ require (
...
-	github.com/rs/zerolog v1.27.0 // indirect
+	github.com/rs/zerolog v1.29.1 // indirect
 	github.com/sasha-s/go-deadlock v0.3.1 // indirect
 	github.com/shirou/gopsutil v3.21.4-0.20210419000835-c7a38de76ee5+incompatible // indirect
 	github.com/spf13/afero v1.9.2 // indirect
 	github.com/spf13/jwalterweatherman v1.1.0 // indirect
 	github.com/spf13/viper v1.14.0 // indirect
 	github.com/status-im/keycard-go v0.0.0-20200402102358-957c09536969 // indirect
 	github.com/subosito/gotenv v1.4.1 // indirect
-	github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7 // indirect
```
Comment (mmsqe) [inline go.mod:178]: need fix [leveldb](https://github.com/crypto-org-chain/cronos/blob/main/go.mod#L242) right?

**Review by mmsqe — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (yihuang) [reply go.mod:178]: yes, fixed.

File: gomod2nix.toml [hunk 4]
```diff
@@ -496,6 +497,7 @@ schema = 3
   [mod."github.com/syndtr/goleveldb"]
     version = "v1.0.1-0.20210819022825-2ae1ddf74ef7"
     hash = "sha256-36a4hgVQfwtS2zhylKpQuFhrjdc/Y8pF0dxc26jcZIU="
+    replaced = "github.com/syndtr/goleveldb"
```
Comment (yihuang) [inline gomod2nix.toml:500]:
```suggestion
```

**Pull request merged**